src/res_core.ml

```diff
@@ -1906,7 +1915,7 @@ and parseFirstClassModuleExpr ~startPos p =
 and parseBracketAccess p expr startPos =
   Parser.leaveBreadcrumb p Grammar.ExprArrayAccess;
   let lbracket = p.startPos in
-  Parser.next p;
+  Parser.expect Lbracket p;
```

Comment:
Though just using `Parser.next` is safe as `p` is always `Lbracket`, this change would make it more clear.